apis/v1beta1/vspherecluster_types.go

```diff
@@ -58,6 +58,12 @@ type VSphereClusterSpec struct {
 	// for each of the objects responsible for creation of VM objects belonging to the cluster.
 	// +optional
 	ClusterModules []ClusterModule `json:"clusterModules,omitempty"`
+
+	// ZoneSelector is the label selector to use for failure domain selection
+	// for the control plane nodes of the cluster.
+	// An empty zone selector includes all the related zones.
+	// +optional
+	ZoneSelector *metav1.LabelSelector `json:"zoneSelector,omitempty"`
```

Comment:
would this be a ZoneSelector or a FailureDomain selector? I think calling it a FailureDomain selector would be clearer that this is tied to a Failure Domain

Reply:
Done, makes sense.